### PR TITLE
Manila liberty config

### PR DESCRIPTION
**Pull request by toabctl**

### CODE REVIEW

**Comment by toabctl** [comment]
Passed minimal manual test (services are running and calling the API works).

**Comment by dirkmueller** [comment]
+1

File: chef/cookbooks/manila/templates/default/manila.conf.erb [hunk 1]
```diff
@@ -1103,54 +1115,78 @@ ssl_key_file=<%= node[:manila][:ssl][:keyfile] if node[:manila][:api][:protocol]
...
+#use_syslog_rfc_format = true
+
+# Syslog facility to receive log lines. This option is ignored if
+# log_config_append is set. (string value)
 #syslog_log_facility = LOG_USER
 
-# Log output to standard error. (boolean value)
+# Log output to standard error. This option is ignored if
+# log_config_append is set. (boolean value)
 #use_stderr = true
```
Comment (vuntz) [inline chef/cookbooks/manila/templates/default/manila.conf.erb]: IIRC, in other components, we set this to false. Should we do the same here?

File: chef/cookbooks/manila/templates/default/manila.conf.erb [hunk 2]
```diff
@@ -1063,6 +1065,16 @@ cinder_admin_auth_url=<%= @keystone_settings['internal_auth_url'] %>
...
 
+# Timeout for client connections socket operations. If an incoming
+# connection is idle for this number of seconds it will be closed. A
+# value of '0' means wait forever. (integer value)
+#client_socket_timeout = 900
+
+# If False, closes the client socket connection explicitly. Setting it
+# to True to maintain backward compatibility. Recommended setting is
+# set it to False. (boolean value)
+#wsgi_keep_alive = true
```
Comment (vuntz) [inline chef/cookbooks/manila/templates/default/manila.conf.erb]: Should we change this as recommended?

File: chef/cookbooks/manila/templates/default/manila.conf.erb [hunk 3]
```diff
@@ -496,13 +368,16 @@ use_syslog=<%= node[:manila][:use_syslog] ? "True" : "False" %>
...
-#osapi_share_listen = 0.0.0.0
+#osapi_share_listen = ::
 osapi_share_listen=<%= @bind_host %>
 
 # Port for OpenStack Share API to listen on. (integer value)
 #osapi_share_listen_port = 8786
 osapi_share_listen_port=<%= @bind_port %>
 
+# Number of workers for OpenStack Share API service. (integer value)
+#osapi_share_workers = 1
```
Comment (vuntz) [inline chef/cookbooks/manila/templates/default/manila.conf.erb]: Don't we want to do the same magic as in other openstack components here?

File: chef/cookbooks/manila/templates/default/manila.conf.erb (hunk 1 above)
Comment (toabctl) [inline chef/cookbooks/manila/templates/default/manila.conf.erb]: done

File: chef/cookbooks/manila/templates/default/manila.conf.erb (hunk 2 above)
Comment (toabctl) [inline chef/cookbooks/manila/templates/default/manila.conf.erb]: done

File: chef/cookbooks/manila/templates/default/manila.conf.erb (hunk 3 above)
Comment (toabctl) [inline chef/cookbooks/manila/templates/default/manila.conf.erb]: done

**Comment by vuntz** [comment]
+1

**Comment by mjura** [comment]
Small nitpick https://github.com/toabctl/crowbar-openstack/commit/915b2c9c26955850187f96d95bdb95220598b12f#diff-5680b338079c3b8170ad708d8207547cR121

I think we can stay with default option.

**Comment by mjura** [comment]
For cinder, we are using nova_catalog_info=compute:nova:internalURL
https://github.com/toabctl/crowbar-openstack/commit/915b2c9c26955850187f96d95bdb95220598b12f#diff-5680b338079c3b8170ad708d8207547cR141

File: chef/cookbooks/manila/templates/default/manila.conf.erb [hunk 4]
```diff
@@ -1769,6 +1923,7 @@ rabbit_host=<%= @rabbit_settings[:address] %>
 rabbit_port=<%= @rabbit_settings[:port] %>
 <% end -%>
 
+
```
Comment (mjura) [inline chef/cookbooks/manila/templates/default/manila.conf.erb]: Small nitpick - this empty line is not needed here

**Comment by mjura** [comment]
Only small nitpicks, but +1

**Comment by mjura** [comment]
BTW. don't we want move all options for backend drivers move to section with multi backend.

**Comment by toabctl** [comment]
> Small nitpick toabctl@915b2c9#diff-5680b338079c3b8170ad708d8207547cR121
> 
> I think we can stay with default option.

@mjura link doesn't work for me.

**Comment by toabctl** [comment]
> For cinder, we are using nova_catalog_info=compute:nova:internalURL
> toabctl@915b2c9#diff-5680b338079c3b8170ad708d8207547cR141

@mjura done in https://github.com/toabctl/crowbar-openstack/commit/8712827cba90c397afa793e8fbfdc4e173bdfe62

**Comment by toabctl** [comment]
> BTW. don't we want move all options for backend drivers move to section with multi backend?

@mjura what do you mean with that?

**Comment by mjura** [comment]
@toabctl  I was thinking about `#nova_catalog_info = compute:nova:publicURL`  for cinder we have `internalURL`

**Comment by mjura** [comment]
@toabctl: With multibackend I meant that for example we have support for netapp and you have defined options for this backend on bottom of `manila.conf.erb` but default options for `netapp_*` are somewhere in middle of config. For cinder.conf.erb we moved everything into 

```
<%
@shares.each_with_index do |share, share_idx|
backend_id = "backend-#{share['backend_driver']}-#{share_idx}"
-%>

[<%= backend_id %>]
....
```

**Comment by toabctl** [comment]
@mjura I think having all backends at the end is more readable. and easier when we update the config for Mitaka

**Comment by mjura** [comment]
ok +1

**Comment by vuntz** [comment]
+1

**Pull request merged**